### PR TITLE
feat: search for `biome` in PATH

### PR DESCRIPTION
**Pull request by nhedger**
This PR introduces a new feature to allow searching for the biome binary in the PATH.

Here's where it comes in the current detection order.

1. Search in `DEBUG_SERVER_PATH`
2. Look in `biome.lspBin`
3. Look in project's dependencies
4. **Search in PATH**
5. Download global binary

### Tests

- [x] macOS
- [x] Linux
- [x] Windows

Possible fix for #127

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -369,14 +370,51 @@ async function getServerPath(
...
 	};
 }
 
+/**
+ * Attempts top resolve the path to the biome binary from the PATH environment variable.
+ *
+ * This runs `which biome` on Unix-like systems and `where biome` on Windows.
+ */
+async function findBiomeInPath(): Promise<string | undefined> {
+	const which = process.platform === "win32" ? "where" : "which";
```
Comment (Zaczero) [inline src/main.ts:408]: Heads up! The `which` program is not always present in all environments. Perhaps it's better to parse $PATH in-code (it's not that complex) without starting an external process and introducing a new dependency.

image

See https://github.com/biomejs/biome-vscode/issues/51#issuecomment-1890876244

**Review by Zaczero — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 1 above)
Comment (nhedger) [reply src/main.ts:408]: Sounds good, I’ll go with your suggestion

**Review by Zaczero — APPROVED** [review]

File: src/main.ts [hunk 2]
```diff
@@ -369,14 +370,56 @@ async function getServerPath(
...
+ * `which` or `where` on the system, or in the PATH.
+ */
+async function findBiomeInPath(): Promise<Uri | undefined> {
+	const path = process.env.PATH;
+
+	if (!path) {
+		return;
+	}
+
+	for (const dir of path.split(delimiter)) {
```
Comment (ematipico) [inline src/main.ts:415]: Shouldn't we split by `:`? `path.delimiter` is `/` or `\`, and `$PATH` uses `:` to delimit the paths.

Comment (nhedger) [reply src/main.ts:415]: `:` is a path _delimiter_
`/` is a path _separator_

https://nodejs.org/api/path.html#pathdelimiter
https://nodejs.org/api/path.html#pathsep

Comment (ematipico) [reply src/main.ts:415]: Oooh, I missed that. Thank you!

**Review by ematipico — APPROVED** [review]

**Pull request merged**